### DISCUSSION TITLE
Docs for merged query defaults

### DISCUSSION THREAD

**Discussion by matt-tingen** [TanStack/query#8485] [General]
I just updated my app from v4 to v5 and was looking to take advantage of the cascading query defaults ([discussion](https://github.com/TanStack/query/discussions/4831), [PR](https://github.com/TanStack/query/pull/4849)). In doing so, I noticed there seems to be a couple of issues in the docs:

- The API docs for `getQueryDefaults`/`setQueryDefaults` still reflect the v4 behavior
- The migration guide makes no mention of the changed behavior

I'd like to make a PR to update the docs, but wanted clarify a couple things:

- The v5 merging behavior is intended to be part of the public API, correct?
- The v4 behavior was documented but having  multiple matching defaults was strongly discouraged by a `console.error`. Because of this, there's arguments to be made both ways as to whether multiple matches were supported in v4 or not. Given this, is the v5 behavior a breaking change or a feature?

**Comment by TkDodo** [comment]
yeah you’re right. I think we changed it but forgot updating the docs. The v5 update was a breaking change I believe.
